src/logic/ask_channel_reporting.ts

```diff
@@ -0,0 +1,77 @@
+import {SectionBlock} from "@slack/web-api";
+import {createSectionBlock, createText} from "../integrations/slack/messages";
+import {getUserProfile} from "../integrations/slack/users";
+import {getDepartmentFromProfile, getDivisionFromProfile, getTeamNameFromProfile} from "../integrations/slack/utils";
+
+export interface AsksChannelReportResult {
+    statsByTeam: Map<string, number>;
+    statsByDivision: Map<string, number>;
+    statsByDepartment: Map<string, number>;
+}
+
+export const createReport = async (messages: any[]) : Promise<AsksChannelReportResult> => {
+    const statsByTeam : Map<string, number> = new Map<string, number>();
+    const statsByDivision :Map<string, number> = new Map<string, number>();
+    const statsByDepartment : Map<string, number> = new Map<string, number>();
+
+    await Promise.all(
+        messages.map(async (message : any) => {
+            const userProfile = await getUserProfile(message.user) || {};
+
+            const teamName: string = getTeamNameFromProfile(userProfile);
+            const divisionName: string = getDivisionFromProfile(userProfile);
+            const departmentName: string = getDepartmentFromProfile(userProfile);
+
+            countEntry(statsByTeam, teamName);
+            countEntry(statsByDivision, divisionName);
+            countEntry(statsByDepartment, departmentName);
+        }));
+
+    // TODO: If I remove this, the sorting gets mixed up. IDK why :|
```

Comment:
This is a weird issue, if anyone got an idea I'd love for some feedback.

Reply:
![CleanShot 2022-05-09 at 15 55 37](https://user-images.githubusercontent.com/10488975/167414390-56c5c52f-d170-45e1-aa40-c789714e211e.png)
This is the 'good' scenario, where input is organized (where the console print is in)

Reply:
![CleanShot 2022-05-09 at 15 56 22](https://user-images.githubusercontent.com/10488975/167414524-5c44c126-d83e-44dd-b56c-ee12677cd33d.png)
This is the faulty scenario, without the console.log.

Notice that in the good situation, 'Rhino' with count of 6 is first, and in the bad scenario 'rhino' with the count of 1 is first.

Reply:
I'm not sure and it's just an idea - I would try changing the code on line 58 to not use the `Symbol.iterator` and generator functions (something like below). I think that might be somehow impacted by lines 31-33 call to `.values()`.

```js
[...collection.entries()].sort((a,b) => a[1] - b[1]).forEach(() => {
... // forEach from line 62
})
```